libafl_tinyinst/src/shim.cc

```diff
@@ -0,0 +1,17 @@
+#include "shim.h"
+
+std::unique_ptr<LiteCov> litecov_new() {
+  return std::make_unique<LiteCov>();
+}
+
+std::unique_ptr<Coverage> coverage_new() {
+  return std::make_unique<Coverage>();
+}
+
+void get_coverage_map(uint8_t *bitmap, size_t map_size, Coverage &newcoverage) {
+  for (auto iter = newcoverage.begin(); iter != newcoverage.end(); iter++) {
```

Comment:
What jackalope does is it ignores coverage for branches that have already been seen before.
That way, `newcoverage` is only filled when we have executed at least one new block.
So actually we can just (somehow? Rust<>CPP interop?) expose the list of new coverage to LibAFL, in something like a `NewCoverageObserver`, and then have a `NewCoverageFeedback` that always returns true if the count is at least one.

Reply:
oh ya i think what they do is have two coverage. one for overall bb and other for new coverage then they call merge function to merge it.